onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.h

```
@@ -0,0 +1,63 @@
+    const auto ret = (val);                                               \
+    ORT_ENFORCE(                                                          \
+        ret == ANEURALNETWORKS_NO_ERROR,                                  \
+        std::string("Error in ") + __FILE__ + std::string(":") +          \
+            std::to_string(__LINE__) + std::string(", function name: ") + \
+            std::string(__func__) + "error, ret: " + GetErrorCause(ret)); \
+  }
+
+#define THROW_ON_ERROR_WITH_NOTE(val, note)                               \
+  {                                                                       \
+    const auto ret = (val);                                               \
+    ORT_ENFORCE(                                                          \
+        ret == ANEURALNETWORKS_NO_ERROR,                                  \
+        std::string("Error in ") + __FILE__ + std::string(":") +          \
+            std::to_string(__LINE__) + std::string(", function name: ") + \
+            std::string(__func__) + "error, ret: " + GetErrorCause(ret) + \
+            std::string(", ") + (note));                                  \
+  }
+
+template <class Map, class Key>
+inline bool Contains(Map map, Key key) {
+  return map.find(key) != map.end();
+}
+
+inline std::string GetErrorCause(int errorCode) {
+  switch (errorCode) {
+    case ANEURALNETWORKS_NO_ERROR:
+      return "No error";
+    case ANEURALNETWORKS_OUT_OF_MEMORY:
+      return "Out of memory";
+    case ANEURALNETWORKS_INCOMPLETE:
+      return "Incomplete";
+    case ANEURALNETWORKS_UNEXPECTED_NULL:
+      return "Unexpected null";
+    case ANEURALNETWORKS_BAD_DATA:
+      return "Bad data";
+    case ANEURALNETWORKS_OP_FAILED:
+      return "Op failed";
+    case ANEURALNETWORKS_BAD_STATE:
+      return "Bad state";
+    case ANEURALNETWORKS_UNMAPPABLE:
+      return "Unmappable";
+    case ANEURALNETWORKS_OUTPUT_INSUFFICIENT_SIZE:
```

This message is potentially confusing. Could we just return the error code as a string, so `return "ANEURALNETWORKS_UNMAPPABLE";` in this case? That way the user gets about the same amount of info but an error code they can directly go search for.